Controllers/EmployeeController.cs

```diff
@@ -16,12 +16,105 @@ namespace TodoApi.Controllers
     public class EmployeeController : ControllerBase
     {
         private readonly ApiDbContext _context;
+        private readonly string connectionString = "Server=myserver;Database=mydb;User Id=admin;Password=password;";
 
 
         public EmployeeController(ApiDbContext context)
         {
             _context = context;
         }
+        
+       [HttpGet]
+        public async Task<IActionResult> GetEmployeeBlocking()
+        {
+            var employee = _context.Employees.FindAsync(1).Result; // Blocking async call
+            return Ok(employee);
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> SlowGetEmployee()
+        {
+            Thread.Sleep(5000); // Blocking thread in async code
+            var employee = await _context.Employees.FindAsync(1);
+            return Ok(employee);
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> GetEmployeeNoDispose()
+        {
+            var context = new ApiDbContext(); // Manual context creation without disposal
+            var employee = await context.Employees.FindAsync(1);
+            return Ok(employee);
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> DoTheThing() // Bad practice: Unclear method name
+        {
+            var x = await _context.Employees.FindAsync(1); // Bad practice: Unclear variable name
+            return Ok(x);
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> GetEmployeesWithSql()
+        {
+            var employees = await _context.Employees
+                .FromSqlRaw("SELECT * FROM Employees WHERE CurrentFlag = 1") // Raw SQL in the controller
+                .ToListAsync();
+            return Ok(employees);
+        }
+
+        [HttpPost]
+        public async Task<ActionResult<Employee>> PostEmployeeWithoutValidation(Employee employee)
+        {
+            // No validation of incoming data
+            _context.Employees.Add(employee);
+            await _context.SaveChangesAsync();
+            return CreatedAtAction("GetEmployee", new { id = employee.BusinessEntityID }, employee);
+        }
+
```

Comment:
Avoid making multiple database calls inside a loop to improve performance. Consider fetching all required data in a single query.